### PR TITLE
ignore defaultChromeFlags / PptrDefaultArgs tags

### PR DESCRIPTION
**Pull request by anemer**
## Proposed changes

Added 2 optional parameters (in `wdio.conf.js` / `capabilities`) that allow:
* to launch Google Chrome without the default flags (given in devtools) 
* and/or to launch browser without the default Puppeteer arguments 

webdriverio#5339

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by anemer** [comment]
Haha!
- Jest: "global" coverage threshold for branches **(96%) not met: 95.99%**
- Test Suites: 201 passed, 201 total
- Tests:       1904 passed, 1904 total
- Snapshots:   89 passed, 89 total
- Time:        133.787s
- Ran all test suites.

**Comment by anemer** [comment]
It's possible.
 But what if someone wants to change some of default Chrome flags, but wants the default Puppeteer's arguments to be preserved (for Chrome or other browsers)?

**Comment by christian-bromann** [comment]
The user would just need to set:

```js
ignorePuppeteerDefaultArgs: {
   // default args to be ignored
   args: [...]
}
```

**Comment by anemer** [comment]
Ok. I will set a single parameter that affects the 2 lists of default args/flags.
It will be exactly like the Puppeteer's `ignoreDefaultArgs` parameter (same name, same type = _<[boolean]|[Array]<[string]>>_)  , and its value will be passed to Puppeteer.

File: packages/devtools/src/launcher.js [hunk 1]
```diff
@@ -37,9 +38,15 @@ async function launchChrome (capabilities) {
         }
     }
 
+    const chromeDefaultFlags = []
+    if (!ignoreDefaultArgs)
+        chromeDefaultFlags.push(...DEFAULT_FLAGS)
+    else if (Array.isArray(ignoreDefaultArgs))
+        chromeDefaultFlags.push(...DEFAULT_FLAGS.filter(flag => !ignoreDefaultArgs.includes(flag)))
```
Comment (christian-bromann) [inline packages/devtools/src/launcher.js:45]: This can be a one liner:

```js
const chromeDefaultFlags = ignoreDefaultArgs && Array.isArray(ignoreDefaultArgs)
    ? ...DEFAULT_FLAGS.filter(flag => !ignoreDefaultArgs.includes(flag))
    : DEFAULT_FLAGS
```

File: packages/devtools/src/launcher.js [hunk 2]
```diff
@@ -60,10 +67,13 @@ async function launchChrome (capabilities) {
...
     log.info(`Launch Google Chrome with flags: ${chromeFlags.join(' ')}`)
-    const chrome = await launchChromeBrowser({
+
+    const chrome = await launchChromeBrowser(Object.assign({
         chromePath: chromeOptions.binary,
         chromeFlags
-    })
+    },
+    ignoreDefaultArgs ? { ignoreDefaultArgs: ignoreDefaultArgs } : {}
+    ))
```
Comment (christian-bromann) [inline packages/devtools/src/launcher.js:76]: No need to use Object.assign:

```js
    const chrome = await launchChromeBrowser({
        chromePath: chromeOptions.binary,
        chromeFlags,
        ignoreDefaultArgs: Boolean(ignoreDefaultArgs)
    })
```

File: packages/devtools/src/launcher.js [hunk 3]
```diff
@@ -90,8 +100,9 @@ async function launchChrome (capabilities) {
     return browser
 }
 
-function launchBrowser (capabilities, product) {
+function launchBrowser(capabilities, product) {
```
Comment (christian-bromann) [inline packages/devtools/src/launcher.js:103]: unnecessary change
```suggestion
function launchBrowser (capabilities, product) {
```

File: packages/devtools/src/launcher.js [hunk 4]
```diff
@@ -109,7 +120,8 @@ function launchBrowser (capabilities, product) {
             width: DEFAULT_WIDTH,
             height: DEFAULT_HEIGHT
         }
-    }, capabilities[vendorCapKey] || {})
+    }, capabilities[vendorCapKey],
+    ignoreDefaultArgs ? { ignoreDefaultArgs: true } : {})
```
Comment (christian-bromann) [inline packages/devtools/src/launcher.js:124]: same as above:

```js
    const puppeteerOptions = Object.assign({
        product,
        executablePath,
        defaultViewport: {
            width: DEFAULT_WIDTH,
            height: DEFAULT_HEIGHT
        },
		ignoreDefaultArgs: Boolean(ignoreDefaultArgs)
    }, capabilities[vendorCapKey] || {})
```

File: packages/devtools/src/launcher.js [hunk 5]
```diff
@@ -119,7 +131,7 @@ function launchBrowser (capabilities, product) {
     return puppeteer.launch(puppeteerOptions)
 }
 
-export default function launch (capabilities) {
+export default function launch(capabilities) {
```
Comment (christian-bromann) [inline packages/devtools/src/launcher.js:134]: unnecessary change
```suggestion
export default function launch (capabilities) {
```

File: packages/devtools/src/launcher.js (hunk 4 above)
Comment (anemer) [reply packages/devtools/src/launcher.js:124]: > Could you move `ignoreDefaultArgs` into `goog:chromeOptions` as it i a custom capability?

It's true my first need was for Chrome. But:
* As I can see in [devtools/launcher.js](https://github.com/webdriverio/webdriverio/blob/master/packages/devtools/src/launcher.js#L129) devtools _can_ also be used for other browsers than Chrome. Maybe not yet at the same level as with Chrome, but it's possible.

* And it seems Puppeteer accepts this `ignoreDefaultArgs` parameter in its experimental support of Firefox (see [experimental/puppeteer-firefox/Launcher.js](https://github.com/puppeteer/puppeteer/blob/v3.0.2/experimental/puppeteer-firefox/lib/Launcher.js#L88)).

so we limit this parameter only for Chrome?

Comment (anemer) [reply packages/devtools/src/launcher.js:124]: > same as above:
> 
> ```js
>     const puppeteerOptions = Object.assign({
>         product,
>         executablePath,
>         defaultViewport: {
>             width: DEFAULT_WIDTH,
>             height: DEFAULT_HEIGHT
>         },
> 		ignoreDefaultArgs: Boolean(ignoreDefaultArgs)
>     }, capabilities[vendorCapKey] || {})
> ```

line 124 is actually wrong. It should be like line 75:
```js
    ignoreDefaultArgs ? { ignoreDefaultArgs: ignoreDefaultArgs } : {}
```
as `ignoreDefaultArgs` is either a _boolean_ or a _array of string_ (as in [Puppeteer](https://pptr.dev/#?product=Puppeteer&version=v3.0.2&show=api-puppeteerlaunchoptions))

File: packages/devtools/src/launcher.js [hunk 6]
```diff
@@ -60,10 +62,13 @@ async function launchChrome (capabilities) {
     }
 
     log.info(`Launch Google Chrome with flags: ${chromeFlags.join(' ')}`)
-    const chrome = await launchChromeBrowser({
+
+    const chrome = await launchChromeBrowser(Object.assign({
```
Comment (christian-bromann) [inline packages/devtools/src/launcher.js:66]: We still don't need `Object.assign` here if `ignoreDefaultArgs` can contain the same values as in Puppeteer we can just do:

```js
const chrome = await launchChromeBrowser({
    chromePath: chromeOptions.binary,
    ignoreDefaultArgs,
    chromeFlags
})
```

File: packages/devtools/src/launcher.js [hunk 7]
```diff
@@ -109,7 +114,8 @@ function launchBrowser (capabilities, product) {
             width: DEFAULT_WIDTH,
             height: DEFAULT_HEIGHT
         }
-    }, capabilities[vendorCapKey] || {})
+    }, capabilities[vendorCapKey],
+    ignoreDefaultArgs ? { ignoreDefaultArgs: ignoreDefaultArgs } : {})
```
Comment (christian-bromann) [inline packages/devtools/src/launcher.js:118]: we can do the same here:

```js
    const puppeteerOptions = Object.assign({
        product,
        executablePath,
        ignoreDefaultArgs,
        defaultViewport: {
            width: DEFAULT_WIDTH,
            height: DEFAULT_HEIGHT
        }
    }, capabilities[vendorCapKey] || {})
```
